cdp_scrapers/legistar_utils.py

```diff
@@ -109,3 +180,858 @@ def get_legistar_events_for_timespan(
+            ev_minutes_item.matter.title = ev_minutes_item.minutes_item.description
+
+        if ev_minutes_item.matter and not ev_minutes_item.matter.result_status:
+            if ev_minutes_item.votes and legistar_ev_item[LEGISTAR_MATTER_STATUS]:
+                # means did not find matter_*_pattern in Legistar EventItemMatterStatus
+                # default to in progress (as opposed to adopted or rejected)
+                # NOTE: if our matter_*_patterns ARE "complete",
+                #       this clause would hit only because the info from Legistar
+                #       is incomplete or malformed
+                ev_minutes_item.matter.result_status = MatterStatusDecision.IN_PROGRESS
+
+        return ev_minutes_item
+
+    def filter_event_minutes(
+        self, ev_minutes_item: EventMinutesItem
+    ) -> EventMinutesItem:
+        """
+        Return None if minutes_item.name contains unimportant text
+        that we want to ignore
+
+        Parameters
+        ----------
+        ev_minutes_item : EventMinutesItem
+
+        Returns
+        -------
+        EventMinutesItem | None
+
+        See Also
+        --------
+        IGNORED_MINUTE_ITEMS
+        """
+        if not ev_minutes_item.minutes_item or not ev_minutes_item.minutes_item.name:
+            return ev_minutes_item
+
+        for filter in self.IGNORED_MINUTE_ITEMS:
+            if filter.lower() in ev_minutes_item.minutes_item.name.lower():
+                return None
+
+        return ev_minutes_item
+
+    def get_none_if_empty(self, model: IngestionModel) -> IngestionModel:
+        """
+        Check required keys in model, return None if any such key has no value.
+        i.e. If all required keys have valid value, return as-is
+
+        Parameters
+        ----------
+        model :
+            Person, MinutesItem, etc.
+
+        Returns
+        -------
+        IngestionModel | None
+            None or model as-is
+        """
+        try:
+            min_keys = self.min_ingestion_keys[model.__class__]
+        except AttributeError:
+            # first time using min_ingestion_keys
+            self.min_ingestion_keys = {}
+            min_keys = None
+        except KeyError:
+            # first time checking model.__class__
+            min_keys = None
+
+        if min_keys is None:
+            min_keys = self.get_required_attrs(model)
+            # cache so we don't do expensive dynamic checking
+            # again for this IngestionModel
+            self.min_ingestion_keys[model.__class__] = min_keys
+
+        if not min_keys:
+            # no required keys for this model
+            # this probably never happens
+            return model
+
+        for key in min_keys:
+            try:
+                val = getattr(model, key)
+
+                # int of 0 is not "empty"
+                if not val and not isinstance(val, int):
+                    # empty value for this key in model
+                    return None
```

Comment:
Why not reverse this and just say `if val is None`. Why check for int?

Reply:
My temporary working version I changed it to:
```python
for k in keys:
            try:
                if getattr(model, k) is None:
                    # bad value for this key in this model
                    return None
            except AttributeError:
                return None

        return model
```

Reply:
My intent was to catch not only `None`, but also "" (empty string), [] (empty list). Because I felt that those are still "unacceptable" for these required fields. So if catching `None` only is fine, then for sure I think this is cleaner and easier to read/maintain.

Reply:
Ohhhhhh nah that makes sense and I like it more. Nice! I would just add your above comment to the code in a bit cleaned up.

"Catch all None and None-like values (empty string, empty list, etc)"